conformance/lang/expressions/let-expr/let_expr.balt

```diff
@@ -0,0 +1,593 @@
+Labels: let-expr, nil-literal, nil-type, boolean, string, boolean-literal, function-call-expr, list-constructor-expr
+
+function init() {
+    io:println(let null n = () in [n, n, n]); // @output [null,null,null]
+    io:println(let null n = () in [n, [n, n]]); // @output [null,[null,null]]
+    io:println(let boolean b1 = true, boolean b2 = false in [b1, b2, b1]); // @output [true,false,true]
+    io:println(let boolean b1 = true, boolean b2 = false in [b2, [b1, b2]]); // @output [false,[true,false]]
+    var c = let string str = "ABCDEF" in str[1];
+    io:println(c); // @output B
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, nil-literal, nil-type, boolean, boolean-literal, function-call-expr, mapping-constructor-expr
+
+function init() {
+    io:println(let null nx = (), null ny = () in {"n": nx, n1: nx, ny, ["a"]: [nx, ny]}); // @output {"n":null,"n1":null,"ny":null,"a":[null,null]}
+    io:println(let null nx = (), null ny = () in {"n": {nx, ny}}); // @output {"n":{"nx":null,"ny":null}}
+    io:println(let boolean b1 = true, boolean b2 = false in {"b": b1, b1: b1, b2, ["a"]: [b1, b2]}); // @output {"b":true,"b1":true,"b2":false,"a":[true,false]}
+    io:println(let boolean b1 = true, boolean b2 = false in {"b": {b1, b2}}); // @output {"b":{"b1":true,"b2":false}}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, nil-literal, nil-type, record-type, union-type, int, function-call-expr, mapping-constructor-expr
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    Record r = let null n = () in {i: n};
+    io:println(r); // @output {"i":null}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, boolean, boolean-literal, union-type, record-type, int, function-call-expr, mapping-constructor-expr
+
+type Record record {|
+    int|boolean? i;
+|};
+
+function init() {
+    Record r = let boolean b = false in {i: b};
+    io:println(r); // @output {"i":false}
+    io:println(let boolean b1 = true, boolean b2 = !b1 in {"b": {b1, b2}}); // @output {"b":{"b1":true,"b2":false}}
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a mapping-constructor-expr.
+Labels: let-expr, string, record-type, array-type, mapping-constructor-expr
+
+type Names record {|
+    string country;
+    string[] names;
+|};
+
+function init() {
+    record {| string myField; |} r = let string smiley = "h😀llo" in {myField: smiley};
+    io:println(r.myField); // @output h😀llo
+}
+
+Test-Case: output
+Description: Test let-expr when expression after `in` is a table-constructor-expr.
+Labels: let-expr, string, boolean, boolean-literal, record-type, function-call-expr, table-constructor-expr
+
+type Names record {|
+    string country;
+    string[] names;
```

Comment:
Missing the array-type label. Please recheck labels.

Reply:
Changed